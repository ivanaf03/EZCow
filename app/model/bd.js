import * as SQLite from "expo-sqlite";

let db;

export const getDatabase = async () => {
  if (!db) {
    db = await SQLite.openDatabaseAsync("ezcow.db");
  }
  return db;
};

export const initDatabase = async () => {
  db = await getDatabase();

  await db.execAsync(`
        PRAGMA journal_mode = WAL;

        CREATE TABLE IF NOT EXISTS User (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            email TEXT NOT NULL UNIQUE,
            password TEXT DEFAULT NULL,
            google_id TEXT DEFAULT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            CHECK (password IS NOT NULL OR google_id IS NOT NULL)
        );

        CREATE TABLE IF NOT EXISTS Cow (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            code TEXT NOT NULL UNIQUE,
            name TEXT NOT NULL,
            entryDate DATE NOT NULL,
            exitDate DATE DEFAULT NULL,
            gender TEXT NOT NULL,
            breed TEXT NOT NULL,
            phase TEXT NOT NULL,
            user_fk INTEGER NOT NULL,
            mother_fk INTEGER DEFAULT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id),
            FOREIGN KEY(mother_fk) REFERENCES Cow(id)
        );

        CREATE TABLE IF NOT EXISTS Field (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            user_fk INTEGER NOT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id)
        );

        CREATE TABLE IF NOT EXISTS BreedingEvent (
            if INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            description TEXT,
            date DATE NOT NULL,
            cow_fk INTEGER NOT NULL,
            FOREIGN KEY(cow_fk) REFERENCES Cow(id)
        );
    `);
};

export const deleteDatabase = async () => {
  await db.closeAsync();
  await SQLite.deleteDatabaseAsync("ezcow.db");
};

export const closeDatabase = async () => {
  await db.closeAsync();
  db = null;
};

export const syncDatabase = async () => {
  await db.execAsync(`
        PRAGMA wal_checkpoint(FULL);
    `);
};
