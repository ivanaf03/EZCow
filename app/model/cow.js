import * as SQLite from 'expo-sqlite';

export const insertCow = async (code, name, entryDate, gender, breed, phase, user_fk, mother_fk) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.runAsync(
        `INSERT INTO Cow (code, name, entryDate, gender, breed, phase, user_fk, mother_fk) VALUES (?, ?, ?, ?, ?, ?, ?, ?)`,
        [code, name, entryDate, gender, breed, phase, user_fk, mother_fk]
    );

    return res;
};

export const getCowIdByCode = async (code) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.getFirstAsync(
        `SELECT id FROM Cow WHERE code = ?`,
        [code]
    );

    return res.id;
};

export const getAvailableCowNameById = async (id) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.getFirstAsync(
        `SELECT name FROM Cow WHERE id = ? AND exitDate is NULL`,
        [id]
    );

    return res.name;
};

export const getAvailableCowCodeByUserId = async (userId) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.getAllAsync(
        `SELECT code FROM Cow WHERE user_fk = ? and exitDate is NULL`,
        [userId]
    );

    return res.map((cow) => cow.code);
};

export const getAllCowsAvailableByUserId = async (userId) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL`,
        [userId]
    );

    return res;
};

export const getFemaleCowsAvailableByUserId = async (userId) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND gender = 'Femenino' AND phase = 'Adulto'`,
        [userId]
    );

    return res;
};

export const getMaleCowsAvailableByUserId = async (userId) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND gender = 'Masculino' AND phase = 'Adulto'`,
        [userId]
    );

    return res;
};

export const getCalvesAvailableByUserId = async (userId) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND phase = 'Ternero'`,
        [userId]
    );

    return res;
};

export const setExitDateByCowId = async (cowId) => {
    const db = await SQLite.openDatabaseAsync('ezcow.db');

    const res = await db.runAsync(
        `UPDATE Cow SET exitDate = CURRENT_DATE WHERE id = ?`,
        [cowId]
    );

    return res;
};
