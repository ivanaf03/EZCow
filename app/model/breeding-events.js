import { getDatabase } from "./bd";

export const insertBreedingEvent = async (
  cow_fk,
  description,
  date,
  user_fk
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, description, date, user_fk) VALUES (?, ?, ?, ?)`,
      [cow_fk, description, date, user_fk]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getBreedingEventsByDayAnUserId = async (day, userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE date = ? AND user_fk = ?`,
      [userId, day]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};