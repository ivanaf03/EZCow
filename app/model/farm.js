import { getDatabase } from "./bd";

export const insertFarmUbication = async (userId, latitude, longitude) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE User SET latitude = ?, longitude = ? WHERE id = ?`,
      [latitude, longitude, userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFarmUbicationByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT latitude, longitude FROM User WHERE id = ?`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
