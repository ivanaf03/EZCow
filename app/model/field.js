import { getDatabase } from "./bd";

export const insertField = async (name, latitude, longitude, user_fk) => {
  const db = await getDatabase();
  try {
    await db.runAsync(
      `INSERT INTO Field (name, latitude, longitude, user_fk) VALUES (?, ?, ?, ?)`,
      [name, latitude, longitude, user_fk]
    );
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllFieldsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(`SELECT * FROM Field WHERE user_fk = ?`, [
      userId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
