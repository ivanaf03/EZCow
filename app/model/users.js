import CryptoJS from "crypto-js";
import { getDatabase } from "./bd";

const hashPassword = (password) => {
  return CryptoJS.SHA256(password).toString();
};

export const insertUser = async (name, email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `INSERT INTO User (name, email, password) VALUES (?, ?, ?)`,
      [name, email, hashedPassword]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertUserGoogle = async (name, email, google_id) => {
  const db = await getDatabase();
  try {
    const existingUser = await db.getFirstAsync(
      `SELECT * FROM User WHERE email = ?`,
      [email]
    );
    if (existingUser) {
      return existingUser;
    }
    const res = await db.runAsync(
      `INSERT INTO User (name, email, google_id) VALUES (?, ?, ?)`,
      [name, email, google_id]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getUserByEmail = async (email) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(`SELECT * FROM User WHERE email = ?`, [
      email,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const changePassword = async (email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `UPDATE User SET password = ? WHERE email = ?`,
      [hashedPassword, email]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
