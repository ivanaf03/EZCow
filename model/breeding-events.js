import { getDatabase } from "../model/bd";

export const insertBreedingEvent = async (
  cow_fk,
  eventName,
  description,
  date
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, eventName, description, date) VALUES (?, ?, ?, ?)`,
      [cow_fk, eventName, description, date]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllBreedingEventsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE user_fk = ? and exitDate IS NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};