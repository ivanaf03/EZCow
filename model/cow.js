import { getDatabase } from "../model/bd";

export const insertCow = async (
  code,
  name,
  entryDate,
  gender,
  breed,
  phase,
  user_fk,
  mother_fk
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO Cow (code, name, entryDate, gender, breed, phase, user_fk, mother_fk) VALUES (?, ?, ?, ?, ?, ?, ?, ?)`,
      [code, name, entryDate, gender, breed, phase, user_fk, mother_fk]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getCowIdByCode = async (code) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(`SELECT id FROM Cow WHERE code = ?`, [
      code,
    ]);
    return res.id;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAvailableCowNameById = async (id) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT name FROM Cow WHERE id = ? AND exitDate is NULL`,
      [id]
    );
    return res.name;
    } catch (error) {
      console.log(error);
      return null;
    }
};

export const getAvailableCowCodeByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT code FROM Cow WHERE user_fk = ? AND exitDate is NULL AND gender='Femenino' AND phase='Adulto'`,
      [userId]
    );
    return res.map((cow) => cow.code);
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAvailableCodeByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT code FROM Cow WHERE user_fk = ? AND exitDate is NULL`,
      [userId]
    );
    return res.map((cow) => cow.code);
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllCowsAvailableByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllCowsAvailableByUserIdAndPhaseAndGender = async (
  userId,
  phase,
  gender
) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND phase = ? AND gender = ?`,
      [userId, phase, gender]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getCalvesAvailableByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND phase = 'Ternero'`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const setExitDateByCowId = async (cowId) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE Cow SET exitDate = CURRENT_DATE WHERE id = ?`,
      [cowId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllCowsExitedByUserIdAndPhaseAndGender = async (
  userId,
  phase,
  gender
) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL AND phase = ? AND gender = ?`,
      [userId, phase, gender]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllCowsExitedByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllCalvesExitedByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL AND phase = 'Ternero'`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllCowIdsAndNamesAvailableByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT id, name FROM Cow WHERE user_fk = ? AND exitDate is NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
