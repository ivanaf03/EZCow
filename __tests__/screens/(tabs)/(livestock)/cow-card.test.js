import React from 'react';
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import CowCard from "../../../../app/(tabs)/(livestock)/cow-card";
import { getAvailableCowNameById, setExitDateByCowId } from '../../../../app/model/cow';

jest.mock("../../../../app/model/cow", () => ({
    getAvailableCowNameById: jest.fn(() => "Vaca Test"),
    setExitDateByCowId: jest.fn(), 
}));

jest.mock("@fortawesome/react-native-fontawesome", () => ({
    FontAwesomeIcon: () => null
}));

describe("CowCard", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", async () => {
        const tree = render(<CowCard cow={{
            id: "1",
            name: "Vaca Test",
            code: "COW001",
            gender: "Femenino",
            breed: "Holstein",
            phase: "Ternero",
            mother_fk: "1234",
            entryDate: "2022-01-01",
        }} />);
        await waitFor (() => expect(tree).toMatchSnapshot());
    });

    it("should render correctly when mother is null", async () => {
        const tree = render(<CowCard cow={{
            id: "1",
            name: "Vaca Test",
            code: "COW001",
            gender: "Femenino",
            breed: "Holstein",
            phase: "Ternero",
            mother_fk: null,
            entryDate: "2022-01-01",
        }} />);
        await waitFor (() => expect(tree).toMatchSnapshot());
    });

    it("should delete cow card on press delete button", async () => {
        const tree = render(<CowCard cow={{
            id: "1",
            name: "Vaca Test",
            code: "COW001",
            gender: "Femenino",
            breed: "Holstein",
            phase: "Ternero",
            mother_fk: "1234",
            entryDate: "2022-01-01",
        }} />);

        await waitFor(() => expect(getAvailableCowNameById).toHaveBeenCalled());
        const deleteButton = tree.getByTestId("delete-cow-button");
        fireEvent.press(deleteButton);
        await waitFor(() => expect(setExitDateByCowId).toHaveBeenCalledWith("1"));
    });
});
