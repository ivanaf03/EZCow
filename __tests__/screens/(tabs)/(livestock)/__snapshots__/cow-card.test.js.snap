// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CowCard should render correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "marginBottom": 8,
        "marginTop": 8,
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "marginLeft": -10,
      },
      {
        "marginRight": 24,
      },
      {
        "backgroundColor": "#ffffff",
      },
      {
        "borderBottomRightRadius": 24,
        "borderTopRightRadius": 24,
      },
      {
        "borderBottomWidth": 4,
        "borderLeftWidth": 4,
        "borderRightWidth": 4,
        "borderTopWidth": 4,
      },
      {
        "borderBottomColor": "#580088",
        "borderLeftColor": "#580088",
        "borderRightColor": "#580088",
        "borderTopColor": "#580088",
      },
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0E041AFF",
            },
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "fontFamily": "Nunito-Bold",
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#a526a9",
            },
            {
              "fontSize": 24,
              "lineHeight": 32,
            },
            {
              "fontFamily": "Nunito-Bold",
            },
            [
              {
                "marginLeft": 16,
              },
            ],
          ]
        }
      >
        Vaca Test
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Crotal:
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontFamily": "Nunito-BoldItalic",
              },
              [
                {
                  "marginLeft": 8,
                },
              ],
            ]
          }
        >
          COW001
        </Text>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Fecha de incorporaci√≥n: 
          1/1/2022
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Sexo: 
          Femenino
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Raza: 
          Holstein
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Fase: 
          Ternero
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Es hijo/a de: 
          Vaca Test
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "alignItems": "center",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "maxWidth": 135,
        },
        [
          {
            "marginTop": 16,
          },
        ],
      ]
    }
    testID="delete-cow-button"
  >
    <Text
      style={
        [
          {
            "color": "#0E041AFF",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
          {
            "fontFamily": "Nunito-Bold",
          },
          [
            {
              "marginLeft": 8,
            },
          ],
        ]
      }
    >
      Dar de baja
    </Text>
  </View>
</View>
`;

exports[`CowCard should render correctly when mother is null 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "marginBottom": 8,
        "marginTop": 8,
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "marginLeft": -10,
      },
      {
        "marginRight": 24,
      },
      {
        "backgroundColor": "#ffffff",
      },
      {
        "borderBottomRightRadius": 24,
        "borderTopRightRadius": 24,
      },
      {
        "borderBottomWidth": 4,
        "borderLeftWidth": 4,
        "borderRightWidth": 4,
        "borderTopWidth": 4,
      },
      {
        "borderBottomColor": "#580088",
        "borderLeftColor": "#580088",
        "borderRightColor": "#580088",
        "borderTopColor": "#580088",
      },
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#0E041AFF",
            },
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "fontFamily": "Nunito-Bold",
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#a526a9",
            },
            {
              "fontSize": 24,
              "lineHeight": 32,
            },
            {
              "fontFamily": "Nunito-Bold",
            },
            [
              {
                "marginLeft": 16,
              },
            ],
          ]
        }
      >
        Vaca Test
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Crotal:
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontFamily": "Nunito-BoldItalic",
              },
              [
                {
                  "marginLeft": 8,
                },
              ],
            ]
          }
        >
          COW001
        </Text>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Fecha de incorporaci√≥n: 
          1/1/2022
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Sexo: 
          Femenino
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Raza: 
          Holstein
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Fase: 
          Ternero
        </Text>
        <Text
          style={
            [
              {
                "color": "#0E041AFF",
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-Bold",
              },
            ]
          }
        >
          Es hijo/a de: 
          Desconocida
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "alignItems": "center",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "maxWidth": 135,
        },
        [
          {
            "marginTop": 16,
          },
        ],
      ]
    }
    testID="delete-cow-button"
  >
    <Text
      style={
        [
          {
            "color": "#0E041AFF",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
          {
            "fontFamily": "Nunito-Bold",
          },
          [
            {
              "marginLeft": 8,
            },
          ],
        ]
      }
    >
      Dar de baja
    </Text>
  </View>
</View>
`;
